### PR TITLE
Followup to #466 - small cleanups for one-time hooks

### PR DESCRIPTION
**Pull request by thockin**
Also includes an e2e cleanup.

@ChrisERo FYI

### CODE REVIEW

File: test_e2e.sh [hunk 1]
```diff
@@ -1109,27 +1096,28 @@ function e2e::exechook_success_once() {
 ##############################################
 function e2e::exechook_fail_once() {
     cat /dev/null > "$RUNLOG"
-        # First sync - return a failure to ensure that we try again
```
Comment (ChrisERo) [inline test_e2e.sh:1112]: My apologies for the bad indentation.

**Review by ChrisERo — COMMENTED** [review]
Looks pretty good to me, just left a nitpick on documentation.

File: pkg/hook/hook.go [hunk 2]
```diff
@@ -98,11 +102,9 @@ type HookRunner struct {
 	data *hookData
 	// Logger
 	logger *logging.Logger
-	// hasCompletedOnce is used to send true if and only if first run executed
-	// successfully and false otherwise to some receiver.  Should be
-	// initialised to a buffered channel of size 1.
-	// Is only meant for use within sendCompletedOnceMessageIfApplicable.
-	hasCompletedOnce chan bool
+	// Used to send a status result when running in one-time mode.
```
[the remark this answers is not in the document]
Comment (thockin) [reply pkg/hook/hook.go:105]: I don't know if we need more - I thought it was pretty self-evident.  For end-users, there will be a log message aabout the failure anyway.

[the remark this answers is not in the document]
Comment (ChrisERo) [reply pkg/hook/hook.go:105]: Sounds good, in that case I'll approve your changes.

**Review by ChrisERo — APPROVED** [review]
LGTM

**Review by stp-ip — APPROVED** [review]
/lgtm

**Pull request merged**